near/src/validator_manager.rs

```diff
@@ -399,7 +458,14 @@ impl ValidatorManager {
             if info.epoch_start_hash != epoch_hash || info.prev_hash == hash {
                 break;
             }
-            proposals.extend(info.proposals);
+            for proposal in info.proposals {
+                if proposal.amount == 0 {
+                    // TODO: check staking key match?
```

Comment:
`account_id` is the right identifier. the same account can't stake twice. only can replace existing key with new key if they send another staking tx.